lib/grpc/GrpcService.ts

```diff
@@ -36,8 +38,21 @@ class GrpcService {
       const response = serializeDateProperties(rawResponse);
       callback(null, response);
     } catch (err) {
+      // if we recognize this error, return a proper gRPC ServiceError with a descriptive and appropriate code
+      let grpcError: ServiceError | undefined;
+      if (err instanceof AssertionError) {
+        grpcError = { name: err.name, message: err.message, code: status.INVALID_ARGUMENT };
```

Comment:
binding `AssertionError` to `INVALID_ARGUMENT` will probably lead to some misleading errors, because we're not limiting the `assert` usage only for arguments validations. So I think we should create a custom, cross-domain error type for that. You can open a separate issue if you don't want to deal with this right now.